### PR TITLE
update task model tests

### LINKED ISSUE #521: Write unit tests for the Task model
Update unit tests for the `Task` Model. These have already been started, but could use more coverage. We should be testing relationship declarations & computed properties (see [testing guide](https://guides.emberjs.com/v2.8.0/testing/testing-models/))

## References
- Progress on: #98 
- [Model testing guide](https://guides.emberjs.com/v2.8.0/testing/testing-models/)
- [Contributing Guidelines](https://github.com/code-corps/code-corps-ember/blob/develop/.github/CONTRIBUTING.md)
- Join our [Slack](https://codecorps.slack.com)

### PR DESCRIPTION
**Pull request by WenInCode**
# What's in this PR?
Update `task` unit model tests

## References
- Fixes #521
- Progress on: #98

### CODE REVIEW

**Comment by begedin** [comment]
I discussed this with @JoshSmith and we think we should sort of look at ember-side model tests as if they were phoenix-side view tests. 

@filipecrosk did this when submitting his PR, by adding a test to assert all required attributes are accounted for. You can see an example at https://github.com/filipecrosk/code-corps-ember/blob/2197e7926fa6f3b1d30eee63a3705623e2a15f18/tests/unit/models/project-test.js#L17

Adding something like this here would probably bee a good idea.

On one hand, we could look at this as "requiring the contributor to add a new attribute to the model twice", but really, we should look at a test like that as a fail-safe to make sure the contributor hasn't accidentally removed a model attribute. Sort of a "you did this, are you sure?".

**Comment by WenInCode** [comment]
I've updated this to test attribute existence. I tried a few methods and the simplest was the `Object.keys` route that @filipecrosk used in his PR.

File: tests/unit/models/task-test.js [hunk 1]
```diff
@@ -17,6 +22,72 @@ test('it exists', function(assert) {
...
+  let task = this.subject();
+  let attributes = Object.keys(task.toJSON());
+
+  assert.ok(attributes.includes('body'), 'task should have the body attribute');
+  assert.ok(attributes.includes('insertedAt'), 'task should have the insertedAt attribute');
+  assert.ok(attributes.includes('likesCount'), 'task should have the likesCount attribute');
+  assert.ok(attributes.includes('markdown'), 'task should have the markdown attribute');
+  assert.ok(attributes.includes('number'), 'task should have the number attribute');
+  assert.ok(attributes.includes('taskType'), 'task should have the taskType attribute');
+  assert.ok(attributes.includes('status'), 'task should have the status attribute');
```
Comment (JoshSmith) [inline tests/unit/models/task-test.js]: Would move up one to stay alpha-ordered.

File: tests/unit/models/task-test.js [hunk 2]
```diff
@@ -17,6 +22,72 @@ test('it exists', function(assert) {
...
+});
+
+test('it should belong to a user', function(assert) {
+  assert.expect(2);
+
+  const task = this.store().modelFor('task');
+  const relationship = get(task, 'relationshipsByName').get('user');
+
+  assert.equal(relationship.key, 'user', 'has relationship with user');
+  assert.equal(relationship.kind, 'belongsTo', 'kind of relationship is belongsTo');
```
Comment (JoshSmith) [inline tests/unit/models/task-test.js]: These tests are starting to seem really duplicative. Wondering if we can instead do something like `assert.hasManyRelationship('task', 'user')` or `assert.belongsToRelationship('task', 'project')` which just wraps everything.

File: tests/unit/models/task-test.js [hunk 3]
```diff
@@ -17,6 +22,72 @@ test('it exists', function(assert) {
   assert.ok(!!model);
 });
 
+test('it should have all of it\'s attributes', function(assert) {
```
Comment (JoshSmith) [inline tests/unit/models/task-test.js]: The `\'` isn't needed here, since it's "its".

**Comment by JoshSmith** [comment]
@WenInCode improved but could use some minor changes first, then you can decide whether to open a separate issue (we'll need one to circle back anyway) or handle here.

File: tests/unit/models/task-test.js (hunk 2 above)
Comment (JoshSmith) [inline tests/unit/models/task-test.js]: In fact, here we go: https://gist.github.com/he9qi/b6354a81a0672dc63294/

File: tests/unit/models/task-test.js (hunk 3 above)
Comment (WenInCode) [inline tests/unit/models/task-test.js]: english 👍

File: tests/unit/models/task-test.js (hunk 2 above)
Comment (WenInCode) [inline tests/unit/models/task-test.js]: @JoshSmith ya that's a good idea. Want to put that in a separate PR? Then I can circle back and update all of these.

**Comment by WenInCode** [comment]
@JoshSmith I brought in those helpers, and addressed remaining feedback. How does this look now?

File: tests/helpers/relationship.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { test } from 'ember-qunit';
+import Ember from 'ember';
+
+// source: https://gist.github.com/he9qi/b6354a81a0672dc63294
+export function testForHasMany(name, many) {
+  test('should have many ' + many, function(assert) {
+    const Model = this.store().modelFor(name);
```
Comment (JoshSmith) [inline tests/helpers/relationship.js]: I would insert `assert.expect(2);` before everything.

File: tests/unit/models/task-test.js [hunk 5]
```diff
@@ -38,55 +34,11 @@ test('it should have all of it\'s attributes', function(assert) {
...
-  const relationship = get(task, 'relationshipsByName').get('taskUserMentions');
-
-  assert.equal(relationship.key, 'taskUserMentions', 'has relationship with taskUserMentions');
-  assert.equal(relationship.kind, 'hasMany', 'kind of relationship is hasMany');
-});
+testForBelongsTo('task', 'project');
+testForBelongsTo('task', 'user');
+testForHasMany('task', 'comments');
+testForHasMany('task', 'commentUserMentions');
+testForHasMany('task', 'taskUserMentions');
```
Comment (JoshSmith) [inline tests/unit/models/task-test.js]: This looks so much better, don't you think? Way easier to write, too.

File: tests/helpers/relationship.js (hunk 4 above)
Comment (WenInCode) [inline tests/helpers/relationship.js]: sure I'll update this

File: tests/unit/models/task-test.js (hunk 5 above)
Comment (WenInCode) [inline tests/unit/models/task-test.js]: Ya it's a lot better

**Comment by JoshSmith** [comment]
Rebase, squash, and merge! 🎉

**Pull request merged**